src/components/ThemeSwitcher.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
+
+import { HiOutlineMoon, HiOutlineSun } from 'react-icons/hi/index';
+
+const ThemeSwitcher = () => {
+  const getInitialState = () => {
+    if (typeof window !== 'undefined') {
+      return localStorage.getItem('theme') as 'light' | 'dark' | undefined;
+    }
+    return 'light';
+  };
+  const [theme, setTheme] = useState<'light' | 'dark' | undefined>(
+    getInitialState
+  );
+
+  const handleThemeSwitch = () => {
+    localStorage.setItem('theme', theme === 'dark' ? 'light' : 'dark');
+    setTheme(theme === 'dark' ? 'light' : 'dark');
+    document.documentElement.classList.toggle('dark');
+  };
+
+  return (
+    <button onClick={handleThemeSwitch} className="flex justify-center">
+      <span className="hidden">{theme}</span>
```

Comment:
It seems like the span here is unnecessary, as it is hidden via CSS.
If it serves any accessibility purpose, let's use `aria-label` on button instead with a more clearer message as `display: none;` also hides from screen readers.

```suggestion
    <button onClick={handleThemeSwitch} className="flex justify-center" aria-label={`Enable ${theme === 'dark' ? 'light' : 'dark'} theme`)>
```